src/coreclr/vm/syncblk.cpp

```diff
@@ -1816,7 +1816,7 @@ BOOL ObjHeader::GetThreadOwningMonitorLock(DWORD *pThreadId, DWORD *pAcquisition
 
             _ASSERTE(psb->GetMonitor() != NULL);
             Thread* pThread = psb->GetMonitor()->GetHoldingThread();
-            if(pThread == NULL)
+            if(pThread == NULL || pThread == (Thread*) -1) // -1 means the lock is orphaned
```

Comment:
This only handled the case of the lock being orphaned when the sync block is created.

It does not handle the case where the lock becomes orphaned after the sync got created.